In [6]:
# import needed library

import pandas as pd
import numpy as np


In [3]:
# Loading data

id_1='1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI'
id_2='1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c'
id_3='10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw'
id_4='1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4'
id_5='1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE'

In [4]:
url_1='https://docs.google.com/spreadsheets/d/' + id_1 + '/export?format=xlsx'
url_2='https://docs.google.com/spreadsheets/d/' + id_2 + '/export?format=xlsx'
url_3='https://docs.google.com/spreadsheets/d/' + id_3 + '/export?format=xlsx'
url_4='https://docs.google.com/spreadsheets/d/' + id_4 + '/export?format=xlsx'
url_5='https://docs.google.com/spreadsheets/d/' + id_5 + '/export?format=xlsx'


In [83]:
df_customer_life=pd.read_excel(url_1, sheet_name='Sheet1')
df_marketing_expenses=pd.read_excel(url_2, sheet_name='Sheet1')
df_monthly_expenses=pd.read_excel(url_3, sheet_name='Sheet1')
df_payroll=pd.read_excel(url_4, sheet_name='Sheet1')
df_payment=pd.read_excel(url_5, sheet_name='Sheet1')


####**Monthly expense**

In [7]:
df_monthly_expenses.head()

#      month           category                  item  amount
0  1 2023-01-01       Server Costs           AWS Hosting    8000
1  2 2023-01-01       Server Costs  Google Cloud Storage    4000
2  3 2023-01-01  Software Licenses        Atlassian Jira    1200
3  4 2023-01-01  Software Licenses                 Slack     800
4  5 2023-01-01  Software Licenses            Salesforce    1500

In [8]:
df_monthly_expenses.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27 entries, 0 to 26
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype         
---  ------    --------------  -----         
 0   #         27 non-null     int64         
 1   month     27 non-null     datetime64[ns]
 2   category  27 non-null     object        
 3   item      27 non-null     object        
 4   amount    27 non-null     int64         
dtypes: datetime64[ns](1), int64(2), object(2)
memory usage: 1.2+ KB


In [9]:
# Create new database for last month expenses:

cond = df_monthly_expenses['month'].dt.strftime('%Y-%m') == '2023-03'

df_last_month_expenses = df_monthly_expenses[cond]
df_last_month_expenses


#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700
23  24 2023-03-01  Software Licenses                  Zoom     540
24  25 2023-03-01      Office Rental           Office Rent   10000
25  26 2023-03-01              Other       Office Supplies     600
26  27 2023-03-01              Other       Travel Expenses    3200

In [10]:
df_monthly_expenses['item'].unique()

array(['AWS Hosting', 'Google Cloud Storage', 'Atlassian Jira', 'Slack',
       'Salesforce', 'Zoom', 'Office Rent', 'Office Supplies',
       'Travel Expenses'], dtype=object)

In [53]:
# Calculate sales & marketing expenses

cond= df_last_month_expenses['item']=='Salesforce'
crm_cost = df_last_month_expenses[cond]['amount'].values[0]

crm_cost

1700

In [54]:
# Add marketing & sales cost to total expenses:

Total_expenses = crm_cost
Total_expenses

1700

In [13]:
type(df_last_month_expenses[cond]['amount'].values)

numpy.ndarray

### **Last month Salary**

In [14]:
# explore dataset

df_payroll.head()

month department  employee_name           position  paid
0 2023-01-01      Sales       John Doe      Sales Manager  1500
1 2023-01-01      Sales     Jane Smith    Sales Associate   600
2 2023-01-01      Sales      Jim Brown    Sales Associate   700
3 2023-01-01      Sales   Laura Miller    Sales Associate   800
4 2023-01-01  Marketing  Alice Johnson  Marketing Manager  1650

In [15]:
df_payroll['yyyy-mm']=df_payroll['month'].dt.strftime('%Y-%m')
df_payroll

month   department   employee_name            position  paid  yyyy-mm
0  2023-01-01        Sales        John Doe       Sales Manager  1500  2023-01
1  2023-01-01        Sales      Jane Smith     Sales Associate   600  2023-01
2  2023-01-01        Sales       Jim Brown     Sales Associate   700  2023-01
3  2023-01-01        Sales    Laura Miller     Sales Associate   800  2023-01
4  2023-01-01    Marketing   Alice Johnson   Marketing Manager  1650  2023-01
5  2023-01-01    Marketing       Bob Davis  Content Specialist   700  2023-01
6  2023-01-01  Engineering  Charlie Wilson    Senior Developer  2500  2023-01
7  2023-01-01  Engineering       Diana Lee           Developer  1700  2023-01
8  2023-01-01  Engineering       Eva White    Junior Developer  1000  2023-01
9  2023-01-01           HR     Frank Green          HR Manager  1300  2023-01
10 2023-01-01           HR      Gary Black       HR Specialist   700  2023-01
11 2023-01-01      Support    Hannah Scott        Support Lead   600  2023-01
12 2023-01-01      Support      Ian Harris  Support Specialist   500  2023-01
13 2023-01-01      Support     Emily Clark  Support Specialist   450  2023-01
14 2023-01-01    Analytics   Michael Brown   Analytics Manager  1700  2023-01
15 2023-01-01    Analytics    Sarah Wilson        Data Analyst  1200  2023-01
16 2023-01-01    Analytics       David Lee        Data Analyst  1300  2023-01
17 2023-02-01        Sales        John Doe       Sales Manager  1500  2023-02
18 2023-02-01        Sales      Jane Smith     Sales Associate   600  2023-02
19 2023-02-01        Sales       Jim Brown     Sales Associate   700  2023-02
20 2023-02-01        Sales    Laura Miller     Sales Associate  5100  2023-02
21 2023-02-01    Marketing   Alice Johnson   Marketing Manager  1650  2023-02
22 2023-02-01    Marketing       Bob Davis  Content Specialist   700  2023-02
23 2023-02-01  Engineering  Charlie Wilson    Senior Developer  2500  2023-02
24 2023-02-01  Engineering       Diana Lee           Developer  1700  2023-02
25 2023-02-01  Engineering       Eva White    Junior Developer  1000  2023-02
26 2023-02-01           HR     Frank Green          HR Manager  1300  2023-02
27 2023-02-01           HR      Gary Black       HR Specialist   700  2023-02
28 2023-02-01      Support    Hannah Scott        Support Lead   600  2023-02
29 2023-02-01      Support      Ian Harris  Support Specialist   500  2023-02
30 2023-02-01      Support     Emily Clark  Support Specialist   450  2023-02
31 2023-02-01    Analytics   Michael Brown   Analytics Manager  1700  2023-02
32 2023-02-01    Analytics    Sarah Wilson        Data Analyst  1200  2023-02
33 2023-02-01    Analytics       David Lee        Data Analyst  1300  2023-02
34 2023-03-01        Sales        John Doe       Sales Manager  1500  2023-03
35 2023-03-01        Sales      Jane Smith     Sales Associate   600  2023-03
36 2023-03-01        Sales       Jim Brown     Sales Associate   700  2023-03
37 2023-03-01        Sales    Laura Miller     Sales Associate   800  2023-03
38 2023-03-01    Marketing   Alice Johnson   Marketing Manager  1650  2023-03
39 2023-03-01    Marketing       Bob Davis  Content Specialist   700  2023-03
40 2023-03-01  Engineering  Charlie Wilson    Senior Developer  2500  2023-03
41 2023-03-01  Engineering       Diana Lee           Developer  1700  2023-03
42 2023-03-01  Engineering       Eva White    Junior Developer  1000  2023-03
43 2023-03-01           HR     Frank Green          HR Manager  1300  2023-03
44 2023-03-01           HR      Gary Black       HR Specialist   700  2023-03
45 2023-03-01      Support    Hannah Scott        Support Lead   600  2023-03
46 2023-03-01      Support      Ian Harris  Support Specialist   500  2023-03
47 2023-03-01      Support     Emily Clark  Support Specialist   450  2023-03
48 2023-03-01    Analytics   Michael Brown   Analytics Manager  1700  2023-03
49 2023-03-01    Analytics    Sarah Wilson        Data Analyst  1200  2023-03
50 2023-03-01    Analytics    

In [39]:
# create function to get last_month data

def get_last_month(df_payroll,month):
  last_month=df_payroll['month'].max().strftime('%Y-%m')

  cond = df_payroll['month'].dt.strftime('%Y-%m') == last_month

  df_last_month_payroll = df_payroll[cond]
  return df_last_month_payroll

df_last_month_payroll = get_last_month(df_payroll,'month')
df_last_month_payroll

month   department   employee_name            position  paid  yyyy-mm
34 2023-03-01        Sales        John Doe       Sales Manager  1500  2023-03
35 2023-03-01        Sales      Jane Smith     Sales Associate   600  2023-03
36 2023-03-01        Sales       Jim Brown     Sales Associate   700  2023-03
37 2023-03-01        Sales    Laura Miller     Sales Associate   800  2023-03
38 2023-03-01    Marketing   Alice Johnson   Marketing Manager  1650  2023-03
39 2023-03-01    Marketing       Bob Davis  Content Specialist   700  2023-03
40 2023-03-01  Engineering  Charlie Wilson    Senior Developer  2500  2023-03
41 2023-03-01  Engineering       Diana Lee           Developer  1700  2023-03
42 2023-03-01  Engineering       Eva White    Junior Developer  1000  2023-03
43 2023-03-01           HR     Frank Green          HR Manager  1300  2023-03
44 2023-03-01           HR      Gary Black       HR Specialist   700  2023-03
45 2023-03-01      Support    Hannah Scott        Support Lead   600  2023-03
46 2023-03-01      Support      Ian Harris  Support Specialist   500  2023-03
47 2023-03-01      Support     Emily Clark  Support Specialist   450  2023-03
48 2023-03-01    Analytics   Michael Brown   Analytics Manager  1700  2023-03
49 2023-03-01    Analytics    Sarah Wilson        Data Analyst  1200  2023-03
50 2023-03-01    Analytics       David Lee        Data Analyst  1300  2023-03

### **Salary of Sales and Marketing specialist**

In [42]:
# Filter data for sales & marketing

last_month_Sales_MKT_payroll=df_last_month_payroll[(df_last_month_payroll['department']=='Sales') | (df_last_month_payroll['department']=='Marketing')]
last_month_Sales_MKT_payroll

month department  employee_name            position  paid  yyyy-mm
34 2023-03-01      Sales       John Doe       Sales Manager  1500  2023-03
35 2023-03-01      Sales     Jane Smith     Sales Associate   600  2023-03
36 2023-03-01      Sales      Jim Brown     Sales Associate   700  2023-03
37 2023-03-01      Sales   Laura Miller     Sales Associate   800  2023-03
38 2023-03-01  Marketing  Alice Johnson   Marketing Manager  1650  2023-03
39 2023-03-01  Marketing      Bob Davis  Content Specialist   700  2023-03

In [55]:
# Calculate salary of sales & marketing

last_month_salary=last_month_Sales_MKT_payroll['paid'].sum()
last_month_salary

5950

In [57]:
# Add salary of sales & marketing to total expenses

Total_expenses+=last_month_salary
Total_expenses

7650

### **Marketing expenses**

In [58]:
# Explore dataset

df_marketing_expenses.head()

date       channel  spending
0 2023-01-01    Google Ads       784
1 2023-01-01  Facebook Ads       659
2 2023-01-01  LinkedIn Ads       729
3 2023-01-01   Twitter Ads       292
4 2023-01-02    Google Ads       935

In [62]:
# create function to get last_month data

def get_last_month(df, date_column):
    last_month = df[date_column].max().strftime('%Y-%m')
    cond = df[date_column].dt.strftime('%Y-%m') == last_month
    df_last_month = df[cond]
    return df_last_month

last_month_mkt_expense=get_last_month(df_marketing_expenses,'date')
last_month_mkt_expense

date       channel  spending
236 2023-03-01    Google Ads       449
237 2023-03-01  Facebook Ads       229
238 2023-03-01  LinkedIn Ads       835
239 2023-03-01   Twitter Ads       986
240 2023-03-02    Google Ads       912
..         ...           ...       ...
355 2023-03-30   Twitter Ads       960
356 2023-03-31    Google Ads       143
357 2023-03-31  Facebook Ads       183
358 2023-03-31  LinkedIn Ads       533
359 2023-03-31   Twitter Ads       909

[124 rows x 3 columns]

In [63]:
# Calculate for last month Marketing expenses

total_mkt_expense = last_month_mkt_expense['spending'].sum()
total_mkt_expense


68830

In [64]:
# Add marketing expenses to total expenses

Total_expenses+=total_mkt_expense
Total_expenses

76480

###**Payment**

In [66]:
# Explore dataset

df_payment.head()

date  customer_id  receipt_amount  new_customer
0 2023-01-01         2653              67             1
1 2023-01-01         2731             271             1
2 2023-01-01         1277             231             0
3 2023-01-01         2094             107             0
4 2023-01-01         1314             416             0

In [67]:
# get last month data

last_month_payment = get_last_month(df_payment,'date')
last_month_payment

date  customer_id  receipt_amount  new_customer
618 2023-03-01         1062             103             0
619 2023-03-01         2243             157             0
620 2023-03-01         1166             372             0
621 2023-03-01         2406             426             1
622 2023-03-01         2761              41             1
..         ...          ...             ...           ...
924 2023-03-31         2079             358             0
925 2023-03-31         2089             375             0
926 2023-03-31         1355             249             0
927 2023-03-31         1062             111             0
928 2023-03-31         1590             271             0

[311 rows x 4 columns]

In [69]:
# Get last month data for new customers

cond=df_payment['new_customer']==1
last_month_payments_from_new_customers=last_month_payment[cond]
last_month_payments_from_new_customers

<ipython-input-69-bf2be6991bed>:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  last_month_payments_from_new_customers=last_month_payment[cond]


date  customer_id  receipt_amount  new_customer
621 2023-03-01         2406             426             1
622 2023-03-01         2761              41             1
625 2023-03-01         2844             252             1
627 2023-03-01         2679             323             1
633 2023-03-02         1475              40             1
..         ...          ...             ...           ...
912 2023-03-29         1746             262             1
913 2023-03-29         2269             484             1
914 2023-03-29         2935             478             1
919 2023-03-29         1297             496             1
920 2023-03-30         2151             482             1

[63 rows x 4 columns]

In [71]:
# Calculate number of new customers:

Number_of_customers=last_month_payments_from_new_customers['customer_id'].nunique()
Number_of_customers

63

##**CAC**

In [73]:
# Calculate last month CAC for new customers:

CAC = Total_expenses/Number_of_customers
CAC

1213.968253968254

###**ARPU**

In [162]:
# Calculate total revenue for new customers

total_revenue = last_month_payments_from_new_customers['receipt_amount'].sum()
total_revenue

17320

In [163]:
# Calculate ARPU

ARPU = total_revenue/Number_of_customers
ARPU

274.92063492063494

##**COGS**

In [82]:
# get last month expense:

last_month_expense = get_last_month(df_monthly_expenses,'month')
last_month_expense

#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700
23  24 2023-03-01  Software Licenses                  Zoom     540
24  25 2023-03-01      Office Rental           Office Rent   10000
25  26 2023-03-01              Other       Office Supplies     600
26  27 2023-03-01              Other       Travel Expenses    3200

In [113]:
# Calculate last month expenses related to product:

last_month_product_expense = last_month_expense[last_month_expense['item'].isin(['Atlassian Jira','Slack','AWS Hosting','Google Cloud Storage'])]
last_month_product_expense

#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900

In [114]:
# Calculate last month product cost

last_month_product_cost=last_month_product_expense['amount'].sum()
last_month_product_cost

15100

In [88]:
df_last_month_payroll.department.unique()

array(['Sales', 'Marketing', 'Engineering', 'HR', 'Support', 'Analytics'],
      dtype=object)

In [119]:
df_last_month_payroll.head(10)

month   department   employee_name            position  paid  yyyy-mm
34 2023-03-01        Sales        John Doe       Sales Manager  1500  2023-03
35 2023-03-01        Sales      Jane Smith     Sales Associate   600  2023-03
36 2023-03-01        Sales       Jim Brown     Sales Associate   700  2023-03
37 2023-03-01        Sales    Laura Miller     Sales Associate   800  2023-03
38 2023-03-01    Marketing   Alice Johnson   Marketing Manager  1650  2023-03
39 2023-03-01    Marketing       Bob Davis  Content Specialist   700  2023-03
40 2023-03-01  Engineering  Charlie Wilson    Senior Developer  2500  2023-03
41 2023-03-01  Engineering       Diana Lee           Developer  1700  2023-03
42 2023-03-01  Engineering       Eva White    Junior Developer  1000  2023-03
43 2023-03-01           HR     Frank Green          HR Manager  1300  2023-03

In [116]:
last_month_salary_product_cost=df_last_month_payroll[(df_last_month_payroll['department']=='Engineering')|(df_last_month_payroll['department']=='Support')]
last_month_salary_product_cost

month   department   employee_name            position  paid  yyyy-mm
40 2023-03-01  Engineering  Charlie Wilson    Senior Developer  2500  2023-03
41 2023-03-01  Engineering       Diana Lee           Developer  1700  2023-03
42 2023-03-01  Engineering       Eva White    Junior Developer  1000  2023-03
45 2023-03-01      Support    Hannah Scott        Support Lead   600  2023-03
46 2023-03-01      Support      Ian Harris  Support Specialist   500  2023-03
47 2023-03-01      Support     Emily Clark  Support Specialist   450  2023-03

In [120]:
# Calculate last month labor cost

last_month_labor_cost=last_month_salary_product_cost['paid'].sum()
last_month_labor_cost

6750

In [121]:
# Calculate COGS

COGS = last_month_product_cost + last_month_labor_cost
COGS

21850

###**Gross Margin**

In [122]:
# Get last month payment

last_month_payment = get_last_month(df_payment,'date')
last_month_payment


date  customer_id  receipt_amount  new_customer
618 2023-03-01         1062             103             0
619 2023-03-01         2243             157             0
620 2023-03-01         1166             372             0
621 2023-03-01         2406             426             1
622 2023-03-01         2761              41             1
..         ...          ...             ...           ...
924 2023-03-31         2079             358             0
925 2023-03-31         2089             375             0
926 2023-03-31         1355             249             0
927 2023-03-31         1062             111             0
928 2023-03-31         1590             271             0

[311 rows x 4 columns]

In [123]:
# Calculate last month revenue:

last_month_revenue = last_month_payment['receipt_amount'].sum()
last_month_revenue

83033

In [139]:
# Calculate Gross Margin

Gross_margin= (last_month_revenue-COGS)/last_month_revenue
Gross_margin

0.7368516132140234

##**LTV**

In [135]:
# Calculate customer's lifespan:

df_customer_life['customer_life'] = (df_customer_life['churn_date'] - df_customer_life['start_date']).dt.days


In [136]:
df_customer_life.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Unnamed: 0     100 non-null    int64         
 1   start_date     100 non-null    datetime64[ns]
 2   churn_date     100 non-null    datetime64[ns]
 3   customer_life  100 non-null    int64         
dtypes: datetime64[ns](2), int64(2)
memory usage: 3.2 KB


In [137]:
df_customer_life.head()

Unnamed: 0 start_date churn_date  customer_life
0        1000 2021-11-15 2022-09-14            303
1        1001 2022-04-15 2023-02-16            307
2        1002 2022-10-30 2023-02-04             97
3        1003 2021-08-22 2023-02-07            534
4        1004 2021-08-23 2022-02-02            163

In [166]:
# Calculate average customer life (month)

AVG_customer_life = df_customer_life['customer_life'].mean()/30
AVG_customer_life

9.841333333333333

In [165]:
# Calculate LTV

LTV = ARPU * AVG_customer_life*Gross_margin
LTV

1993.6151202865285

###**LTV/CAC ratio**

In [164]:
# Calculate LTV/CAC ratio

LTV_CAC_ratio = LTV / CAC

LTV_CAC_ratio

1.6422300284787041

#**Result**

In [145]:
# CAC:
print(f'CAC: {CAC}')

CAC: 1213.968253968254


In [146]:
# ARPU:
print(f'ARPU: {ARPU}')

ARPU: 274.92063492063494


In [152]:
# COGS:
print(f'COGS: {COGS}')

COGS: 21850


In [154]:
# Gross Margin:
print(f'Gross Margin: {Gross_margin}')

Gross Margin: 0.7368516132140234


In [155]:
# LTV
print(f'LTV: {LTV}')

LTV: 1993.6151202865285


In [157]:
# LTV/CAC ratio:
print(f'LTV/CAC ratio: {LTV_CAC_ratio}')

LTV/CAC ratio: 1.6422300284787041


#**Conclusion**

**LTV/CAC ratio = 1.6422**

The **LTV/CAC** ratio is **greater than 1**, indicates that TechStream Solutions is generating more revenue from customers than it costs to acquire them. This is a positive sign, suggesting profitability
and sustainable growth.

However, an ideal LTV/CAC ratio typically ranges from 3 to 5, indicating a healthy balance between customer acquisition costs and the value generated from customers. TechStream Solutions shoud try to improve this ratio to ideal range.

The average customer lifespans of TechStream Solution is only 9.81 months, company should find solutions to improve this lifespan in order to imporve LTV and LTV/CAC ratio.

